### PR TITLE
[llvm] Fix disabling direct calls and inlining

### PR DESCRIPTION
**Pull request by alexanderkyte**
Previously we disabled too much runtime format when we disable direct calls. This fixes that.

By putting inlining behind a different flag than direct calls, you can tell if it's the direct call itself or the inlining it enables that leads to a specific bit of observed behavior.

### CODE REVIEW

**Comment by akoeplinger** [comment]
I can confirm that with `llvm-disable-direct-calls` MONO_DEBUG option set it fixes the slow LLVM issue from https://github.com/mono/mono/issues/14843.

**Comment by akoeplinger** [comment]
The latest version works without the `llvm-disable-direct-calls` option now :+1:

**Comment by akoeplinger** [comment]
@monojenkins backport to 2019-06

File: mono/mini/mini-runtime.c [hunk 1]
```diff
@@ -3731,7 +3733,7 @@ mini_parse_debug_options (void)
...
+			fprintf (stderr, "Available options: 'handle-sigint', 'keep-delegates', 'reverse-pinvoke-exceptions', 'collect-pagefault-stats', 'break-on-unverified', 'no-gdb-backtrace', 'suspend-on-native-crash', 'suspend-on-sigsegv', 'suspend-on-exception', 'suspend-on-unhandled', 'dont-free-domains', 'dyn-runtime-invoke', 'gdb', 'explicit-null-checks', 'gen-seq-points', 'no-compact-seq-points', 'single-imm-size', 'init-stacks', 'casts', 'soft-breakpoints', 'check-pinvoke-callconv', 'use-fallback-tls', 'debug-domain-unload', 'partial-sharing', 'align-small-structs', 'native-debugger-break', 'thread-dump-dir=DIR', 'no-verbose-gdb', 'llvm_disable_inlining', 'llvm-disable-self-init'.\n");
```
Comment (jaykrell) [inline mono/mini/mini-runtime.c]: It’d help diff/merge/readability if this line was split up.

Comment (alexanderkyte) [reply mono/mini/mini-runtime.c]: Or kept in sync with the options list in a non-manual way

**Comment by marek-safar** [comment]
@vargaz, please review

**Comment by vargaz** [comment]
Shouldn't this be an option to llc/opt ?

**Comment by alexanderkyte** [comment]
It's not an opt/llc argument because it's a mono-side feature to toggle on and off (whether to use the replace_all_uses_of at the end there). I think you meant as an option to "--aot", which is a good idea that was considered in channel when this feature was first merged. 

@vargaz the argument that was made in favor of MONO_DEBUG is that it's a bit harder to use it to check if the subsystem is working by toggling it on/off if you have to change args to the AOT compiler in the build files, rather than just exporting a symbol in the environment outside the build system, and having the child subprocess inherit it.

It's not all that much harder though. Could definitely also have the option to live in both ```--aot``` and MONO_DEBUG if you want that.

**Pull request merged**